specutils/utils/wcs_utils.py

```diff
@@ -550,7 +550,14 @@ def gwcs_from_array(array):
     Create a new WCS from provided tabular data. This defaults to being
     a GWCS object.
     """
-    array = u.Quantity(array)
+    orig_array = u.Quantity(array)
+
+    # TODO: Input arrays must be strictly ascending. This is not always the
+    #  case for a spectral axis (e.g. when in frequency space). Thus, we
+    #  convert to wavelength to create the wcs.
```

Comment:
Maybe add the note here that the reason it has to be ascending is because the way that `astropy.modeling.tabular.Tabular1D` is coded up to use `scipy.interpolate`.  The `inverse` for `Tabular1D` already handles this problem, but the forward should as well.  See https://github.com/astropy/astropy/blob/d6a069ee3e5831b58a83ce3e0a5bbf6a7aeaef4f/astropy/modeling/tabular.py#L238-L248